elytron/src/main/java/org/wildfly/extension/elytron/DomainDefinition.java

```diff
@@ -550,5 +557,155 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
+                .setAllowExpression(false)
+                .build();
+
+        static void register(ManagementResourceRegistration resourceRegistration, ResourceDescriptionResolver descriptionResolver) {
+            resourceRegistration.registerOperationHandler(new SimpleOperationDefinition(ElytronDescriptionConstants.READ_IDENTITY, descriptionResolver, NAME), new ReadSecurityDomainIdentityHandler());
+        }
+
+        private ReadSecurityDomainIdentityHandler() {
+        }
+
+        @Override
+        protected void executeRuntimeStep(final OperationContext context, final ModelNode operation) throws OperationFailedException {
+            // Retrieving the modifiable registry here as the SecurityDomain is used to create a new authentication
+            // context.
+            ServiceRegistry serviceRegistry = context.getServiceRegistry(true);
+            RuntimeCapability<Void> runtimeCapability = SECURITY_DOMAIN_RUNTIME_CAPABILITY.fromBaseCapability(context.getCurrentAddressValue());
+            ServiceName domainServiceName = runtimeCapability.getCapabilityServiceName(SecurityDomain.class);
+            ServiceController<SecurityDomain> serviceController = getRequiredService(serviceRegistry, domainServiceName, SecurityDomain.class);
+            SecurityDomain domain = serviceController.getValue();
+            ServerAuthenticationContext authenticationContext = domain.createNewAuthenticationContext();
+            String principalName = NAME.resolveModelAttribute(context, operation).asString();
+
+            try {
+                authenticationContext.setAuthenticationName(principalName);
+
+                if (!authenticationContext.exists()) {
+                    context.getFailureDescription().add(ROOT_LOGGER.identityNotFound(principalName));
+                    return;
+                }
+
+                if (!authenticationContext.authorize(principalName)) {
+                    context.getFailureDescription().add(ROOT_LOGGER.identityNotAuthorized(principalName));
+                    return;
+                }
+
+                SecurityIdentity identity = authenticationContext.getAuthorizedIdentity();
+                ModelNode result = context.getResult();
+
+                result.get(ElytronDescriptionConstants.NAME).set(principalName);
+
+                ModelNode attributesNode = result.get(ElytronDescriptionConstants.ATTRIBUTES);
+
+                identity.getAttributes().entries().forEach(entry -> {
+                    ModelNode entryNode = attributesNode.get(entry.getKey()).setEmptyList();
+                    entry.forEach(entryNode::add);
+                });
+
+                ModelNode rolesNode = result.get(ElytronDescriptionConstants.ROLES);
+                identity.getRoles().forEach(rolesNode::add);
+            } catch (RealmUnavailableException e) {
+                throw ROOT_LOGGER.couldNotReadIdentity(principalName, domainServiceName, e);
```

Comment:
Don't throw OFE unless this is a user mistake; throw RuntimeException for server faults.

If it's unclear, it's not the end of the world to throw OFE.